### PR TITLE
Dynamic font per service

### LINKED ISSUE #1745: Make embedded fonts dynamic per service
**Is your feature request related to a problem? Please describe.**
As we move to more services we need to ensure the correct fonts are on the correct services

**Describe the solution you'd like**
Create a `fonts` array in each service context, and import in the desired fonts. This can then be passed into `<GlobalStyle />` via the service context provider.

| Service | Required fonts |
|:--------|:---------------|
| news    | `F_REITH_SERIF_MEDIUM, F_REITH_SERIF_MEDIUM_ITALIC, F_REITH_SANS_REGULAR, F_REITH_SANS_ITALIC, F_REITH_SANS_BOLD, F_REITH_SANS_BOLD_ITALIC` |
| persian | none |
| igbo    | none  |
| thai    | none  |
| pidgin  | none  |

**Testing notes**
[Tester to complete]

Dev insight: Unit tests, maybe a cypress test?

**Additional context**
Will confirm with Articles what fonts persian need, as I imagine they wont need italic

### PR DESCRIPTION
**Pull request by ghost**
Resolves #1745

**Overall change:** Add fonts config to services to be used on `ServiceContext`

**Code changes:**

- Add fonts array to each service config
- Add service fonts through globalStyle
- Update `withPageWrapper` component and test.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/Layouts/defaultPageWrapper.jsx [hunk 1]
```diff
@@ -10,14 +10,16 @@ import ConsentBanner from '../containers/ConsentBanner';
...
 
 const PageWrapper = ({ bbcOrigin, children, id, service, isAmp }) => {
   const env = process.env.APP_ENV;
   const { isUK, origin } = getOriginContext(bbcOrigin);
+  const { fonts } = services[service];
 
   return (
     <Fragment>
-      <GlobalStyle />
+      <GlobalStyle fonts={fonts} />
```
Comment (dr3) [inline src/app/Layouts/defaultPageWrapper.jsx]: If you move `<GlobalStyle fonts={fonts} />` to inside the children of `ServiceContextProvider`, `GlobalStyle` can use context to get the fonts, which I think is nicer than accessing the config directly

File: src/app/containers/PageHandlers/withPageWrapper.jsx [hunk 2]
```diff
@@ -4,15 +4,16 @@ import articlePropTypes from '../../models/propTypes/article';
 import PageWrapper from '../../Layouts/defaultPageWrapper';
 
 const WithPageWrapper = Component => {
-  const PageWrapperContainer = props => (
-    <PageWrapper {...props}>
-      <Component {...props} />
+  const PageWrapperContainer = ({ data, bbcOrigin, service }) => (
```
Comment (dr3) [inline src/app/containers/PageHandlers/withPageWrapper.jsx]: Why cant this just use `props` like it did before?

File: src/app/lib/globalStyles.js [hunk 3]
```diff
@@ -21,14 +13,9 @@ const GlobalStyle = createGlobalStyle`
...
-
-  ${F_REITH_SERIF_MEDIUM}
-  ${F_REITH_SERIF_MEDIUM_ITALIC}
-  ${F_REITH_SANS_REGULAR}
-  ${F_REITH_SANS_ITALIC}
-  ${F_REITH_SANS_BOLD}
-  ${F_REITH_SANS_BOLD_ITALIC}
-`;
+  ${({ fonts }) =>
+    fonts.reduce((fontStyles, fontStyle) => fontStyles + fontStyle, '')}
```
Comment (dr3) [inline src/app/lib/globalStyles.js]: Whats this doing?

File: src/app/lib/__snapshots__/globalStyles.test.js.snap [hunk 4]
```diff
@@ -15,66 +15,13 @@ Array [
...
-  }",
-  "
-  @font-face {
-    font-family: \\"ReithSans\\";
-    src: url(\\"https://gel.files.bbci.co.uk/r2.302/BBCReithSans_W_BdIt.woff2\\") format(\\"woff2\\"), url(\\"https://gel.files.bbci.co.uk/r2.302/BBCReithSans_W_BdIt.woff\\") format(\\"woff\\");
-    font-weight: 700;
-    font-style: italic;
-    font-display: optional;
-  }",
+  [Function],
```
Comment (dr3) [inline src/app/lib/__snapshots__/globalStyles.test.js.snap]: Can we get some better test coverage for this file please? Weve lost the snap coverage so its not actually testing it atm

**Review by dr3 — DISMISSED** [review]

**Review by j-pendlebury — APPROVED** [review]
Nice! 😄

**Review by andrew-nowak — APPROVED** [review]
:+1:

**Comment by david-boydell** [comment]
This has been verified and is rfm.

**Pull request merged**